tests/core/test_actions.py

```diff
@@ -762,4 +757,61 @@ def test_get_form_action_if_not_in_forms():
     )
 
     with pytest.raises(ActionNotFoundException):
-        assert not action.action_for_name(form_action_name, domain, None)
+        assert not action.action_for_name_or_text(form_action_name, domain, None)
+
+
+@pytest.mark.parametrize(
+    "end_to_end_utterance", ["Hi", f"{UTTER_PREFIX} is a dangerous start"]
+)
+def test_get_end_to_end_utterance_action(end_to_end_utterance: Text):
+    domain = Domain.from_yaml(
+        f"""
+    actions:
+    - my_action
+    {KEY_E2E_ACTIONS}:
```

Comment:
should use the actual string rather than a constant

Reply:
@wochinge could you please take this over

Reply:
What's the reason for not using a constant? 
* In my opinion it increases the maintainability of the tests as we treat is as production code
* If we don't expect a constant to change ever, then hardcoding it in tests can be ok in my opinion.
* I'm not sure how it affects readability tbh. On the one hand it might be easier to read, on the other hand the constant makes it easier to establish a context

Reply:
I'd expect a tests on the API to break if I change constants that are used to define the API. @mprazz might jump in with some QA foo, but I think in general it is a good practice to keep the tests separated from the code and constants under test.

Reply:
👍🏻 
To me this means we lose half of the advantage of constants, no? If a value is actually part of the API then I'd rather do the following:

1) Use a constant in the test code
2) Add a test which ensure that a constant has a certain value

This way we get the best out of both:
* Maintainability
* Sure a change to the API changes a test